airflow/www/static/js/grid/api/useGridData.js

```diff
@@ -67,9 +64,14 @@ const useGridData = () => {
   }, {
     // only refetch if the refresh switch is on
     refetchInterval: isRefreshOn && autoRefreshInterval * 1000,
-    initialData,
-    placeholderData: emptyData,
   });
+
+  // Placeholder object that is not use on refetch, but only
+  // for the initial load.
+  if (!isSuccess) {
```

Comment:
Why did you make this change?

Reply:
`InitialData` is populating the cache for all key, therefore successive changes to the filters do not trigger an api call. (Even if the query key changed). We have to wait for `queryClient.staleTime` to pass, defined in our config.  This is the expected behavior, I linked in the PR description the related discussion.

It looks like placeholder data is supplied each time you are "refeching" for another key that is not present in cache -> this induced a flickering with old data on each refetch.